ocs_ci/helpers/storage_auto_scaler.py

```diff
@@ -100,25 +101,116 @@ def wait_for_auto_scaler_status(
+    if scaling_threshold - min_diff < max_used_capacity:
         logger.info(
             f"The scaling_threshold {scaling_threshold} is too close to the used "
-            f"capacity {used_capacity}. Increasing the scaling_threshold"
+            f"capacity {max_used_capacity}. Increasing the scaling_threshold."
         )
-        scaling_threshold = int(used_capacity) + min_diff
+        scaling_threshold = int(max_used_capacity) + min_diff
 
     return scaling_threshold
+
+
+def verify_autoscaler_status_not_trigger(auto_scaler_name, namespace=None):
+    """
+    Verify that the StorageAutoscaler does not enter 'InProgress' and remains NotStarted.
+
+    Args:
+        auto_scaler_name (str): Name of the StorageAutoScaler resource
+        namespace (str): Namespace of the autoscaler. Defaults to ENV_DATA namespace.
+    """
+    namespace = namespace or config.ENV_DATA["cluster_namespace"]
+
+    verify_not_trigger = False
+    timeout = constants.PROMETHEUS_RECONCILE_TIMEOUT
+    logger.info(f"Wait {timeout} seconds to verify the autoscaler doesn't trigger.")
+    try:
+        wait_for_auto_scaler_status(
+            expected_status=constants.IN_PROGRES,
+            namespace=namespace,
+            resource_name=auto_scaler_name,
+            timeout=timeout,
+            sleep=30,
+        )
+    except TimeoutExpiredError:
+        logger.info(
+            "The StorageAutoScaler didn't reach the InProgress status as expected"
+        )
+        verify_not_trigger = True
+
+    assert verify_not_trigger, "The StorageAutoScaler has been triggered"
+
+    storage_auto_scaler = OCP(kind=constants.STORAGE_AUTO_SCALER, namespace=namespace)
+    auto_scaler_status = storage_auto_scaler.get_resource_status(
+        auto_scaler_name, "PHASE"
+    )
+    assert (
+        auto_scaler_status == constants.NOT_STARTED
+    ), f"The StorageAutoScaler is not in the expected status {constants.NOT_STARTED}"
+
+
+def safe_teardown_delete_all_autoscalers(namespace=None):
+    """
+    Safely delete all StorageAutoScaler resources, ensuring Prometheus has reconciled since the last cleanup.
+
+    This function performs the following:
+    - Reads the last cluster cleanup timestamp from `config.RUN["cleanup_cluster_time"]`
+    - If set, waits for a Prometheus reconcile timeout (default: 660 seconds) from that time
+    - Then deletes all autoscaler resources in the given namespace
+
+    This helps prevent premature scaling or leftover metrics issues caused by fast teardown/recreate cycles.
+
+    Args:
+        namespace (str): Namespace of the autoscaler. Defaults to ENV_DATA namespace.
+
+    """
+    namespace = namespace or config.ENV_DATA["cluster_namespace"]
+
+    cleanup_cluster_time = config.RUN.get("cleanup_cluster_time")
+    if not cleanup_cluster_time:
+        logger.info("No last cleanup cluster time recorded — skipping wait.")
+        delete_all_storage_autoscalers(namespace)
+        return
+
+    time_remaining = (
```

Comment:
could it be <0 in some edge cases?

Reply:
Yes. If everything works as expected, the cleanup is executed at the end of the test, before the teardown. For example, if we collect logs before the teardown and it takes more than 10 minutes, the `time_remaining` will be less than 0.